hudi-common/src/main/java/org/apache/hudi/common/table/read/KeyBasedFileGroupRecordBuffer.java

```diff
@@ -100,17 +102,8 @@ public void processDeleteBlock(HoodieDeleteBlock deleteBlock) throws IOException
     Iterator<DeleteRecord> it = Arrays.stream(deleteBlock.getRecordsToDelete()).iterator();
     while (it.hasNext()) {
       DeleteRecord record = it.next();
-      processNextDeletedRecord(record, record.getRecordKey());
-    }
-  }
-
-  @Override
-  public void processNextDeletedRecord(DeleteRecord deleteRecord, Serializable recordKey) {
-    BufferedRecord<T> existingRecord = records.get(recordKey);
```

Comment:
I don't think it makes sense to merge the handling of `processNextDataRecord` and `processNextDeletedRecord `

Reply:
> There can be deletions inside of data records as well 

it's true but the paload data is there while the `DeleteRecord` is not, I know unifying of code is good but it woud mess up the `BufferedRecord` -> `HoodieRecord` conversion because the later can only be constructed as empty hoodie record.

Reply:
I don't understand why this would mess up the conversion. How is it different than what is in place today where we handle comparing BufferedRecords that are deletes to BufferedRecords that are not deletes?